src/lib/event_xml_parser.c

```diff
@@ -386,6 +386,7 @@ static void text(GMarkupParseContext *context,
 
                         log_info("event name:'%s'", text_copy);
                         ec_set_screen_name(ui, text_copy);
+                        free(text_copy);
                         text_copy = NULL;
```

Comment:
You can use [`g_clear_pointer()`](https://developer.gnome.org/glib/stable/glib-Memory-Allocation.html#g-clear-pointer) for all the instances of the free-then-clear pattern.